### PR TITLE
Run tests on more pypy versions in CI

### PR DESCRIPTION
**Pull request by AlexWaygood**
An attempt to make breakages like #258 less likely. I haven't included pypy-3.7, since our tests are currently failing on pypy-3.7 (#258). For the third-party tests we run, I've only included pypy in the matrix if the upstream package actually tests with pypy (and I've only included the specific pypy versions that they test with)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,7 +41,21 @@ jobs:
...
+          - "3.8"
+          - "3.8.0"
+          - "3.9"
+          - "3.9.0"
+          - "3.10"
+          - "3.10.0"
+          - "3.11"
+          - "3.11.0"
+          - "3.12"
+          - "pypy3.8"
```
Comment (AlexWaygood) [inline .github/workflows/ci.yml:56]:
```suggestion
```

File: .github/workflows/third_party.yml [hunk 2]
```diff
@@ -41,7 +41,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "pypy3.8", "pypy3.9"]
```
Comment (AlexWaygood) [inline .github/workflows/third_party.yml:44]:
```suggestion
        python-version: ["3.8", "3.9", "3.10", "3.11", "pypy3.9"]
```

File: .github/workflows/third_party.yml [hunk 3]
```diff
@@ -307,7 +308,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "pypy3.8"]
```
Comment (AlexWaygood) [inline .github/workflows/third_party.yml:311]:
```suggestion
        python-version: ["3.8", "3.9", "3.10", "3.11", "pypy3.9"]
```

**Comment by AlexWaygood** [comment]
Our tests are also failing on pypy-3.8, it looks like, so I've also removed that from the matrix.

**Review by srittau — APPROVED** [review]

**Pull request merged**